src/async-input.js

```diff
@@ -39,6 +39,27 @@ export class AsyncInput {
   getValue() {}
 }
 
+/**
+ * Attributes
+ *
+ * Components implementing the AsyncInput,
+ * are expected to assert these attributes
+ * at the appropriate time.
+ *
+ * @constant {!Object<string, string>}
```

Comment:
This is an `@enum {string}` and you can reference it with dot-notation. Closure Compiler should mangle the property name, but not the value.

Reply:
cc @zhouyx Didn't know we could do this! 😄 

Thanks @cvializ !

Reply:
yes! `@enums` are great since I think they compile down to just the `string` value, which lets Closure Compiler get rid of the object reference.